### ISSUE TITLE
Loading packages from the "base" env after `TestEnv.activate` (and useage with ReTest.jl)

### ISSUE THREAD

**Issue by nickrobinson251** [JuliaTesting/TestEnv.jl#19]
I have ReTest.jl in my "base" (v1.6) environment, but it seems that after `TestEnv.activate()` i'm no longer in an env that has ReTest 

e.g.
```julia
julia> Pkg.activate("~/repos/ChainRules.jl/Project.toml")
  Activating environment at `~/repos/ChainRules.jl/Project.toml`

julia> using ReTest  # works

julia> Base.LOAD_PATH
3-element Vector{String}:
 "@"
 "@v#.#"
 "@stdlib"

julia> TestEnv.activate()
"/var/folders/hx/1h0bbkfd18d4n1qrnwmrl4j00000gn/T/jl_240blx/Project.toml"

julia> using ReTest  # oops
ERROR: ArgumentError: Package ReTest not found in current path:
- Run `import Pkg; Pkg.add("ReTest")` to install the ReTest package.

Stacktrace:
 [1] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:893
   
julia> Base.LOAD_PATH
2-element Vector{String}:
 "@"
 "/var/folders/hx/1h0bbkfd18d4n1qrnwmrl4j00000gn/T/jl_240blx"
```

**Comment by oxinabox** [comment]
It is this line
https://github.com/JuliaTesting/TestEnv.jl/blob/8d8c90cce003e58f3c8f216757707a4fcaae7876/src/activate_set.jl#L53

It needs to just be
```julia
  push!(LOAD_PATH, tmp)
```

Also this lines too should probably be modified similarly
https://github.com/JuliaTesting/TestEnv.jl/blob/8d8c90cce003e58f3c8f216757707a4fcaae7876/src/activate_set.jl#L83


Dropping the global enviroment during `]test` makes sense, but not in `TestEnv.activate()`


Probably to fix I will create a `release-1.6` based on `release-1.7`
and make that change.
Then port that to 1.7 and 1.8 also

**Comment by oxinabox** [comment]
This is now closed in 1.7 and 1.8,
but backport is pending for 1.4-1.6
I think it isn't a proble for 1.0->1.3 because they use a different mechanism
